wicket-core/src/main/java/org/apache/wicket/markup/html/form/Form.java

```diff
@@ -1711,7 +1725,7 @@ public final void writeHiddenFields()
 	{
 		// if it's a get, did put the parameters in the action attribute,
 		// and have to write the url parameters as hidden fields
-		if (encodeUrlInHiddenFields())
+		if (METHOD_POST.equalsIgnoreCase(getMethod()))
```

Comment:
Sven Meier comment from https://issues.apache.org/jira/browse/WICKET-6807?focusedCommentId=17172167&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17172167:

That is not correct!

Parameters **must** be written as hidden fields, when the Form is 'get' (because the browser would strip them from the form's action). For a 'post' Form the parameters **can** be either in hidden fields **or** in the form's action, both would work.

Reply:
In that case let's always write them as hidden fields.

Reply:
I've updated it.
I want to check what else could be simplified by using always hidden fields. I don't see a reason to write extra query parameters in the form action url instead of using hidden fields everywhere.